### ISSUE TITLE
mix local.hex does not work with Erlang 23 (httpc error - :malformed_handshake_data)

### ISSUE THREAD

**Issue by michaelst** [elixir-lang/elixir#9889]
This may be because Elixir is just not ready to work with Erlang 23, but just thought I would bring this up as I haven't seen any mentions of it anywhere.

### Environment

* Elixir & Erlang/OTP versions (elixir --version): 

Erlang/OTP 23 [RELEASE CANDIDATE 1] [erts-11.0] [source] [64-bit] [smp:6:6] [ds:6:6:10] [async-threads:1]

Elixir 1.10.2 (compiled with Erlang/OTP 23)

* Operating system: 

Alpine Linux (running in docker)

### Current behavior
```
mix local.hex --force

04:36:02.530 [info]  TLS :client: In state :certify at tls_connection.erl:1281 generated CLIENT ALERT: Fatal - Handshake Failure
 - :malformed_handshake_data
** (Mix) httpc request failed with: {:failed_connect, [{:to_address, {'repo.hex.pm', 443}}, {:inet, [:inet], {:tls_alert, {:handshake_failure, 'TLS client: In state certify at tls_connection.erl:1281 generated CLIENT ALERT: Fatal - Handshake Failure\n malformed_handshake_data'}}}]}

Could not install Hex because Mix could not download metadata at https://repo.hex.pm/installs/hex-1.x.csv.
```
### Expected behavior

Successfully install hex.

**Comment by josevalim** [comment]
Hi @michaelst! This has been reported to Erlang/OTP, so we are waiting for a fix. But yeah, let's leave it open so we don't forget to track it. Thanks!

**Comment by vans163** [comment]
Same issue just now using master branch OTP and master elixir.

**Comment by michaelst** [comment]
This is fixed in RC2

**Comment by mroach** [comment]
As this issues still exists in the latest OTP, 23.0.3, anyone finding this issue and needs a quick albeit insecure workaround:

```shell
env HEX_UNSAFE_HTTPS=1 mix deps.get
```

**Comment by dmorneau** [comment]
I previously reported hex being broken on OTP 23-rc3, but I verified this was fixed in the official 23.0 release. It sounds like you’re hitting a different problem?

https://bugs.erlang.org/plugins/servlet/mobile#issue/ERL-1237

**Comment by mroach** [comment]
@dmorneau Sorry, you're right, I'm hitting a different problem with our internal hex repo:

```shell
Request failed ({:failed_connect, [{:to_address, {'hex.our-internal-domain.io', 443}}, {:inet, [:inet], {:tls_alert, {:handshake_failure, 'TLS client: In state certify at ssl_handshake.erl:1840 generated CLIENT ALERT: Fatal - Handshake Failure\n {bad_cert,hostname_check_failed}'}}}]})
** (Mix) Package fetch failed and no cached copy available (https://hex.our-internal-domain.io/tarballs/rest_api-0.3.5.tar)
```
